### PR TITLE
[WIP] Expose liveness and service response times metrics through Prometheus sidecar

### PR DESCRIPTION
**Pull request by Daniel-Pivonka**
#371  Initial work of metrics collection.

For now grpc response times are collected and written the log and put in a Prometheus end point in the form of a histogram.

### CODE REVIEW

File: pkg/csi-common/metrics.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (Madhu-1) [inline pkg/csi-common/metrics.go]: need to change the year

File: pkg/csi-common/metrics.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+// InitMetrics func
+func InitMetrics() {
+	klog.V(3).Infof("Init Metrics Collection")
+
+	http.Handle("/metrics", promhttp.Handler())
+
+	go func() {
+		if err := http.ListenAndServe(":8080", nil); err != http.ErrServerClosed {
```
Comment (Madhu-1) [inline pkg/csi-common/metrics.go]: am not sure how feasible is running two processes inside a single container

how we can check the status of two services to which one is working fine?
@Daniel-Pivonka is it possible to expose these metrics as an RPC method and run a Prometheus sidecar container to collect these metrics?

**Comment by Madhu-1** [comment]
@Daniel-Pivonka planning to work on this?

**Comment by Daniel-Pivonka** [comment]
> @Daniel-Pivonka planning to work on this?

Not during this sprint.

**Comment by humblec** [comment]
@Daniel-Pivonka this can be closed , Isnt it ?

**Comment by Daniel-Pivonka** [comment]
yes closing

**Pull request closed without merging**